engine/src/main/java/org/hibernate/validator/internal/metadata/core/MetaConstraint.java

```diff
@@ -41,13 +45,28 @@
 	private final ConstraintLocation location;
 
 	/**
+	 * The {@link ValueExtractor} used to extract the value for validation.
+	 */
+	private final ValueExtractorDescriptor valueExtractorDescriptor;
```

Comment:
Nope. `ConstraintLocation` is the object defining a location, it can be shared by several different constraints, some of them requiring unwrapping, some of them not.